tests/utils/client.go

```diff
@@ -91,6 +151,25 @@ func (c *KubernetesTestClient) WaitForStatefulSetReadyReplicasCount(name, namesp
 	}
 }
 
+func (c *KubernetesTestClient) CheckIfPodExists(name, namespace string) bool {
+	pod := c.GetPod(name, namespace)
+	if pod == nil {
+		log.Warningf("Found 0 pods with name %s in namespace %s .", name, namespace)
+		return false
+	}
+	log.Infof("Found 1 pod with name %s in %s namespace", name, namespace)
+	return true
+}
+
+func (c *KubernetesTestClient) GetPod(name, namespace string) *v1.Pod {
+	pod, err := c.CoreV1().Pods(namespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		log.Warningf("%v", err)
+		return nil
```

Comment:
let's return an `error` here and make the method return `(*v1.Pod, error)`
and handle it `err != nil` where it's using this method.
returning `nil` is not a good pattern unless the type returned is `error` only. 

https://www.digitalocean.com/community/tutorials/handling-errors-in-go